actionpack/test/dispatch/debug_locks_test.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require "abstract_unit"
+
+class DebugLocksTest < ActionDispatch::IntegrationTest
+  def setup
+    build_app
+  end
+
+  def test_render_threads_status
+    thread = Thread.new do
+      ActiveSupport::Dependencies.interlock.running { sleep 1 }
```

Comment:
Thanks for adding this! Whoever originally added the middleware obviously forgot to include a test... whoever that was. 🙈

Instead of a sleep, we should be able to do this with `Concurrent::*` primitives and/or a mutex.

Reply:
Thanks for the review! I updated to use `Concurrent::CountDownLatch`.